tests/test_search.py

```diff
@@ -71,3 +74,17 @@ def test_full_search(es_client):
 
     for obj in SimpleObject.search_objects.search().full():
         assert isinstance(obj, SimpleObject)
+
+
+@pytest.mark.django_db
+def test_search_next_generator(es_client):
+    management.call_command("sync_es")
+
+    mommy.make(SimpleObject, baz="tired", _quantity=10)
+    mommy.make(SimpleObject, baz="awake", _quantity=10)
+    SimpleObject.search_objects.refresh()
+
+    search = SimpleObject.search_objects.search()
+    for item in search:
+        assert item
+        assert item == six.next(search)
```

Comment:
Oh ok so the for loop calls LazySearch.__getitem__ and six.next() triggers LazySearch.next().